ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/CIOEngine.kt

```diff
@@ -34,38 +34,30 @@ internal class CIOEngine(
     }
 
     override suspend fun execute(data: HttpRequestData): HttpResponseData {
+        val callContext = callContext()!!
+
         while (true) {
-            if (closed.value) throw ClientClosedException()
+            if (closed) {
```

Comment:
Consider checking it in `executeWithinCallContext` once for all engines.

Reply:
Done.